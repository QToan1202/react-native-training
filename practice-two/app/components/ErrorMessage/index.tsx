import { memo } from 'react'
import isEqual from 'react-fast-compare'
import { FieldError } from 'react-hook-form'
import { SizableText, SizableTextProps } from 'tamagui'

import Heading from '@components/Heading'

export type ErrorMessageProps = {
  error: FieldError | undefined
} & SizableTextProps

const ErrorMessage = ({ error, ...rest }: ErrorMessageProps) => (
  <SizableText lineHeight="$4" {...rest}>
    {error && (
      <Heading
        paddingLeft="$space.4"
        color="$color.red"
        fontSize={13}
        content={String(error.message)}
        {...rest}
      />
    )}
  </SizableText>
)

export default memo(ErrorMessage, isEqual)
