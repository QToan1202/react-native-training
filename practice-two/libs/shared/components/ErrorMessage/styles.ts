import { StyleSheet } from 'react-native'

import { COLORS } from '../../constants'

const styles = StyleSheet.create({
  errorWrapper: {
    height: 20,
  },
  error: {
    paddingLeft: 20,
    color: COLORS.RED,
    fontSize: 13,
    lineHeight: 20,
  },
})

export default styles
