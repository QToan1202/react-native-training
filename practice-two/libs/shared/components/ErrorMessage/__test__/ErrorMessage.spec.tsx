import { render } from '@testing-library/react-native'
import { FieldError } from 'react-hook-form'

import ErrorMessage from '../index'

jest.mock('react-hook-form')

const fakeError: FieldError = { type: 'required', message: 'error' }

describe('Testing Error Message component', () => {
  it('Render component successfully', () => {
    const component = render(<ErrorMessage error={fakeError} />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('Message must visible on the screen', () => {
    const { getByText } = render(<ErrorMessage error={fakeError} />)

    expect(getByText(/error/)).toBeVisible()
  })
})
