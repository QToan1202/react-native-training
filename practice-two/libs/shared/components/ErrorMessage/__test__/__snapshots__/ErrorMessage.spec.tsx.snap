// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Error Message component Render component successfully 1`] = `
<View
  style={
    {
      "height": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 24,
        },
        false,
        [
          {
            "color": "#ff5880",
            "fontSize": 13,
            "lineHeight": 20,
            "paddingLeft": 20,
          },
          undefined,
        ],
      ]
    }
  >
    error
  </Text>
</View>
`;
